# gibby
`gibby` is a Python package designed to process genomic data, extract peak sequences, compute positional frequency and weight matrices (PFMs, PWMs), and compare scores based on the PWM in order to converge on the best motif. This tool is particularly for motif finding.

## Input 
`gibby` can take HOMER `peaks.txt` files and `.bed` peak files as input.

## Output
Three files are generated: `PFM.txt`, `PWM.txt`, and `motif.png`. These files include the position frequency matrix and position weight matrix generated by gibbs sampling, and a visualization of the sequence motif showing which motif was most strongly present. 

## What is Gibbs Sampling?
Gibbs Sampling is a statistical method used to estimate the distribution of variables when direct sampling is difficult. It is particularly useful in motif finding where we want to identify common patterns (motifs) in a set of sequences. 
<p align="center">
  <img src="https://github.com/kairitanaka/CSE_185_finalProject/assets/64274901/e9ca348f-9c26-45b7-b3d9-da83b10b6a41" alt="gibbs">
</p>

# For example ...
Imagine you are trying to perfect a secret recipe, but you don't have all the ingredients at once. You start by randomly choosing some ingredients and proportions, then you taste the result. Based on how good or bad it tastes, you keep some ingredients, change others, and try again. Each iteration helps you understand what works and what doesn't, gradually leading you to the best recipe.

In the context of genomic sequences, Gibbs Sampling helps us identify common patterns (motifs) by iteratively refining our guesses based on the sequences we have. Each iteration, which involves some randomness, helps to gradually reveal the underlying motifs more accurately.

More images and text to come explaining gibbs ... 

## Features

- **Compute Position Frequency/Weight Matrices (PFMs/PWMs)**: Generate a PFM/PWM from the extracted peak region sequences.
- **Gibbs Sampling for Motif Finding**: Utilize Gibbs sampling to identify motifs within the peak sequences.

## Installation

You can install the package via pip:

```
pip install git+https://github.com/kairitanaka/CSE_185_finalProject.git
```
Once you run it it will be helpful to run:
```
gibby_motifFinding -h
```
to view all options


## Usage

`gibby_motifFinding` from Gibby (ver 1.0.0) utilizes Gibbs Sampling to find potential motifs that are in peak regions of the genome. The potential motifs are used to generate a position frequency matrix (PFM), a position weight matrix (PWM), and a motif logo based on these matrices.

```
gibby_motifFinding [-h] -f PEAK_FILE -t PEAK_FILE_TYPE -g GENOME_FASTA_FILE [-s SCORE_THRESHOLD] [-k KMER_SIZE] [-i ITERATIONS]
```


### Command Line Arguments

- `-f`, `--peak_file`: Input BED/Homer file with peak information. (required)
- `-t`, `--peak_file_type`: Specify peak file type: 'bed' or 'homer' file. (required)
- `-g`, `--genome_fasta_file`: Genome FASTA file. (required)
- `-s`, `--score_threshold`: Score threshold for filtering peaks. (optional)
- `-k`, `--kmer_size`: Size of k-mers for motif finding. (optional)
- `-i`, `--iterations`: Number of iterations for Gibbs sampler. (optional)



