<!DOCTYPE html>
<!-- saved from url=(0101)https://meme-suite.org/meme/info/status?service=MEMECHIP&id=appMEMECHIP_5.5.51717289802740-2093843272 -->
<html><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
    
    <meta http-equiv="cache-control" content="no-cache">
    <meta http-equiv="pragma" content="no-cache">
    <meta http-equiv="expires" content="Sun 1 Jan 1997 00:00:00 GMT">
    <title>Job Status - MEME Suite</title>
    <link rel="shortcut icon" type="image/x-icon" href="https://meme-suite.org/meme/doc/images/memesuite_icon.ico">
    <link rel="icon" type="image/png" href="https://meme-suite.org/meme/doc/images/memesuite_icon.png">
    <link href="./Job Status - MEME Suite_files/style.css" rel="stylesheet" type="text/css">
    <link href="./Job Status - MEME Suite_files/status.css" rel="stylesheet" type="text/css">
    <script src="./Job Status - MEME Suite_files/menu-path.js.download"></script><style type="text/css" id="operaUserStyle"></style>
    <script src="./Job Status - MEME Suite_files/menu.js.download"></script><link rel="stylesheet" type="text/css" href="./Job Status - MEME Suite_files/menu.css"><script type="text/javascript" src="./Job Status - MEME Suite_files/menu-path(1).js.download"></script><script type="text/javascript" src="./Job Status - MEME Suite_files/Markdown.Converter.js.download"></script><script type="text/javascript" src="./Job Status - MEME Suite_files/menu-data.js.download"></script><script type="text/javascript" src="./Job Status - MEME Suite_files/menu-configure.js.download"></script>
    <script src="./Job Status - MEME Suite_files/utilities.js.download"></script>
    <script src="./Job Status - MEME Suite_files/motif_logo.js.download"></script>
    <script src="./Job Status - MEME Suite_files/site.js.download"></script>
    <script src="./Job Status - MEME Suite_files/status.js.download"></script>
    <script>
      var service = "MEMECHIP";
      var id = "appMEMECHIP_5.5.51717289802740-2093843272";
      var report = 
{
  "items": [
    {
      "type": "sequences",
      "name": "(Primary) Sequences",
      "key": "sequences"
    },
    {
      "type": "sequences",
      "name": "Control Sequences",
      "key": "negSeq"
    },
    {
      "type": "flag",
      "name": "Convert DNA to RNA",
      "key": "dna2rna",
      "on": "Input sequences are DNA but will be converted to RNA."
    },
    {
      "type": "motifs",
      "name": "Motifs",
      "key": "motifs"
    },
    {
      "type": "range",
      "name": "Motif Width",
      "keyLow": "minw",
      "keyHigh": "maxw",
      "same": "MEME and STREME will find motifs which are exactly !!LOW!! wide.",
      "low": "MEME and STREME will find motifs that are at least !!LOW!! wide.",
      "high": "MEME and STREME will find motifs that are at most !!HIGH!! wide.",
      "both": "MEME and STREME will find motifs that are between !!LOW!! and !!HIGH!! wide (inclusive)."
    },
    {
      "type": "background",
      "name": "Background",
      "key": "background"
    },
    {
      "type": "flag",
      "name": "Parse Genomic Coordinates",
      "key": "pgc",
      "on": "Motif sites will be output in genomic coordinates if possible."
    },
    {
      "type": "choice",
      "name": "Discovery Mode",
      "key": "disc_mode",
      "options": {
        "classic": "Classic: optimizes the E-value of the motif information content",
        "de": "Differential Enrichment: optimizes the significance of the relative enrichment of motif sites",
        "psp": "Discriminative: creates a discriminative position-specific prior (PSP) and applies Classic mode"
      }
    },
    {
      "type": "choice",
      "name": "MEME Site Distribution",
      "key": "memeOptMode",
      "options": {
        "zoops": "MEME will only find motifs with zero or one occurrence (of a contributing motif site) per sequence.",
        "oops": "MEME will only find motifs with one occurrence (of a contributing motif site) per sequence.",
        "anr": "MEME will find motifs with any number of repetitions (of a contributing motif site per sequence)."
      }
    },
    {
      "type": "count",
      "name": "MEME Motif Count",
      "key": "memeOptNMotifs",
      "one": "MEME will search for one motif.",
      "any": "MEME will search for !!VALUE!! motifs."
    },
    {
      "type": "range",
      "name": "MEME Motif Sites",
      "keyLow": "memeOptMinSites",
      "keyHigh": "memeOptMaxSites",
      "same": "MEME will find motifs with exactly !!LOW!! sites.",
      "low": "MEME will find motifs with at least !!LOW!! sites.",
      "high": "MEME will find motifs with at most !!HIGH!! sites.",
      "both": "MEME will find motifs with between !!LOW!! and !!HIGH!! sites (inclusive)."
    },
    {
      "type": "flag",
      "name": "MEME Palindromes",
      "key": "memeOptPal",
      "on": "MEME will restrict its search to palindromic motifs."
    },
    {
      "type": "flag",
      "name": "MEME Non-random Subsampling",
      "key": "memeOptNorand",
      "on": "MEME will not randomize the (primary) sequence order before sampling."
    },
    {
      "type": "number",
      "name": "STREME <i>p</i>-value Threshold",
      "key": "stremeOptP",
      "any": "STREME will search for motifs with a <i>p</i>-value &le; !!VALUE!!."
    },
    {
      "type": "count",
      "name": "STREME Motif Count",
      "key": "stremeOptM",
      "one": "STREME will search for one motif.",
      "any": "STREME will search for !!VALUE!! motifs."
    },
    {
      "type" : "choice",
      "name": "STREME Site Positional Distribution Plots",
      "key": "stremeOptAlign",
      "options": {
        "left": "Sequences are aligned on their left ends.",
        "center": "Sequences are aligned on their centers.",
        "right": "Sequences are aligned on their right ends.",
      }
    },
    {
      "type": "number",
      "name": "CentriMo Match Score",
      "key": "centrimoOptScore",
      "any": "CentriMo will only use motif sites if they have a match score &ge; !!VALUE!!."
    },
    {
      "type": "number",
      "name": "CentriMo Region Width",
      "key": "centrimoOptMaxReg",
      "any": "CentriMo will only consider regions if they are &le; !!VALUE!! in width."
    },
    {
      "type": "number",
      "name": "CentriMo <i>E</i>-value Threshold",
      "key": "centrimoOptEThresh",
      "any": "CentriMo will only report motifs which have a region enrichment <i>E</i>-value &le; !!VALUE!!."
    },
    {
      "type": "flag",
      "name": "CentriMo Local",
      "key": "centrimoOptLocal",
      "on": "CentriMo will search for enriched regions anywhere.",
      "off": "CentriMo will only find central enriched regions."
    },
    {
      "type": "flag",
      "name": "CentriMo Store IDs",
      "key": "centrimoOptStoreIds",
      "on": "CentriMo will store the sequence identifiers and their mapping to the best region of all significant motifs."
    }
  ]
};
    </script>
  <style type="text/css">:root [href^="//mage98rquewz.com/"], :root [href^="//x4pollyxxpush.com/"], :root zeus-ad, :root span[id^="ezoic-pub-ad-placeholder-"], :root ins.adsbygoogle[data-ad-slot], :root guj-ad, :root gpt-ad, :root div[id^="zergnet-widget"], :root div[id^="vuukle-ad-"], :root div[id^="taboola-stream-"], :root div[id^="sticky_ad_"], :root div[id^="pa_sticky_ad_box_middle_"], :root div[id^="optidigital-adslot"], :root div[id^="gpt_ad_"], :root div[id^="ezoic-pub-ad-"], :root div[id^="dfp-ad-"], :root div[id^="crt-"][style], :root div[id^="advads_ad_"], :root div[id^="adspot-"], :root div[id^="ads300_250-widget-"], :root div[id^="ads300_100-widget-"], :root div[id^="ads250_250-widget-"], :root div[id^="adrotate_widgets-"], :root ps-connatix-module, :root div[id^="ad_position_"], :root div[id^="ad-div-"], :root div[id^="_vdo_ads_player_ai_"], :root div[id*="ScriptRoot"], :root div[id*="MarketGid"], :root div[data-native_ad], :root div[data-native-ad], :root div[data-mini-ad-unit], :root div[data-insertion], :root div[data-id-advertdfpconf], :root div[data-google-query-id], :root hl-adsense, :root div[data-contentexchange-widget], :root div[data-content="Advertisement"], :root div[data-alias="300x250 Ad 2"], :root div[data-adzone], :root div[data-adunit-path], :root div[data-adname], :root div[data-ad-wrapper], :root div[data-ad-placeholder], :root div[class^="native-ad-"], :root div[data-dfp-id], :root div[class^="kiwi-ad-wrapper"], :root div[aria-label="Ads"], :root display-ads, :root display-ad-component, :root bottomadblock, :root atf-ad-slot, :root aside[id^="adrotate_widgets-"], :root ark-top-ad, :root amp-fx-flying-carpet, :root amp-embed[type="taboola"], :root amp-connatix-player, :root div[id^="google_dfp_"], :root ad-slot, :root a[href^="https://banners.livepartners.com/"], :root a[target="_blank"][onmousedown="this.href^='http://paid.outbrain.com/network/redir?"], :root a[onmousedown^="this.href='https://paid.outbrain.com/network/redir?"][target="_blank"] + .ob_source, :root a[onmousedown^="this.href='http://paid.outbrain.com/network/redir?"][target="_blank"], :root a[href^="https://www.purevpn.com/"][href*="&utm_source=aff-"], :root a[href^="https://www.onlineusershielder.com/"], :root a[href^="https://www.nudeidols.com/cams/"], :root a[href^="https://financeads.net/tc.php?"], :root a[href^="https://www.mrskin.com/tour"], :root a[href^="https://www.kingsoffetish.com/tour?partner_id="], :root a[href^="https://www.infowarsstore.com/"] > img, :root a[href^="https://www.highperformancecpmgate.com/"], :root amp-ad, :root a[href^="https://www.highcpmrevenuenetwork.com/"], :root a[href^="https://lnkxt.bannerator.com/"], :root a[href^="https://www.geekbuying.com/dynamic-ads/"], :root a[href^="https://www.friendlyduck.com/AF_"], :root a[href^="https://www.financeads.net/tc.php?"], :root [href^="https://www.herbanomic.com/"] > img, :root a[href^="https://maymooth-stopic.com/"], :root a[href^="https://www.dql2clk.com/"], :root a[href^="https://www.nutaku.net/signup/landing/"], :root a[href^="https://www.dating-finder.com/signup/?ai_d="], :root a[href^="https://www.brazzersnetwork.com/landing/"], :root a[href^="https://www.adxsrve.com/"], :root [data-template-type="nativead"], :root a[href^="https://www.endorico.com/Smartlink/"], :root a[href^="https://www.adultempire.com/"][href*="?partner_id="], :root a[href^="https://voluum.prom-xcams.com/"], :root a[href^="https://twinrdsrv.com/"], :root a[href^="https://trk.nfl-online-streams.club/"], :root a[href^="https://tracking.avapartner.com/"], :root a[href^="https://track.wg-aff.com"], :root a[href^="https://track.ultravpn.com/"], :root a[href^="https://track.afcpatrk.com/"], :root a[href^="https://track.1234sd123.com/"], :root a[href^="https://torguard.net/aff.php"] > img, :root a[href^="https://tc.tradetracker.net/"] > img, :root a[href^="https://tatrck.com/"], :root a[href^="https://click.candyoffers.com/"], :root [href^="https://zstacklife.com/"] img, :root a[href^="https://t.aslnk.link/"], :root a[href^="https://t.adating.link/"], :root a[href^="https://safesurfingtoday.com/"][href*="?skip="], :root a[href^="https://land.brazzersnetwork.com/landing/"], :root a[href^="https://t.acam.link/"], :root a[href^="https://syndication.optimizesrv.com/"], :root a[href^="https://go.trackitalltheway.com/"], :root [href^="https://track.fiverr.com/visit/"] > img, :root a[href^="https://syndication.exoclick.com/"], :root a[href^="https://syndication.dynsrvtbg.com/"], :root div[data-alias="300x250 Ad 1"], :root a[href^="https://syndicate.contentsserved.com/"], :root a[href^="https://svb-analytics.trackerrr.com/"], :root a[href^="https://streamate.com/landing/click/"], :root a[href^="https://ad.doubleclick.net/"], :root a[href^="https://static.fleshlight.com/images/banners/"], :root a[href^="https://slkmis.com/"], :root a[href^="https://s.zlink3.com/"], :root a[href^="https://www.mrskin.com/account/"], :root a[href^="https://s.optzsrv.com/"], :root a[href^="https://quotationfirearmrevision.com/"], :root a[href^="https://pubads.g.doubleclick.net/"], :root a[href^="https://ak.oalsauwy.net/"], :root a[href^="https://softwa.cfd/"], :root a[href^="https://play1ad.shop/"], :root a[href^="https://prf.hn/click/"][href*="/camref:"] > img, :root a[href^="https://www.dating-finder.com/?ai_d="], :root a[href^="https://serve.awmdelivery.com/"], :root a[href^="https://prf.hn/click/"][href*="/adref:"] > img, :root app-ad, :root a[href^="https://postback1win.com/"], :root a[href^="https://a.adtng.com/"], :root a[href^="https://playnano.online/offerwalls/?ref="], :root a[href^="https://www.privateinternetaccess.com/"] > img, :root a[href^="https://mmwebhandler.aff-online.com/"], :root a[href^="https://www.bet365.com/"][href*="affiliate="], :root a[href^="https://pb-track.com/"], :root a[href^="https://pb-front.com/"], :root a[href^="https://paid.outbrain.com/network/redir?"], :root a[href^="https://ngineet.cfd/"], :root a[href^="https://ndt5.net/"], :root a[href^="http://eighteenderived.com/"], :root a[href^="https://natour.naughtyamerica.com/track/"], :root a[href^="https://mediaserver.entainpartners.com/renderBanner.do?"], :root a[href^="https://www.sugarinstant.com/?partner_id="], :root a[href^="https://m.do.co/c/"] > img, :root .nya-slot[style], :root a[href^="https://a.bestcontentweb.top/"], :root a[href^="https://lobimax.com/"], :root a[href^="https://lead1.pl/"], :root a[href^="https://landing1.brazzersnetwork.com"], :root a[href^="https://landing.brazzersnetwork.com/"], :root a[href^="https://join.virtuallust3d.com/"], :root a[href^="https://kiksajex.com/"], :root a[href^="https://juicyads.in/"], :root a[href^="https://www.mypornstarcams.com/landing/click/"], :root a[href^="https://snowdayonline.xyz/"], :root a[href^="https://mediaserver.gvcaffiliates.com/renderBanner.do?"], :root a[href^="https://join.dreamsexworld.com/"], :root a[href^="https://jaxofuna.com/"], :root a[href^="https://itubego.com/video-downloader/?affid="], :root a[href^="https://italarizege.xyz/"], :root a[href^="https://iqbroker.com/"][href*="?aff="], :root ad-shield-ads, :root a[href^="https://hot-growngames.life/"], :root a[href^="https://golinks.work/"], :root a[href^="https://go.xxxvjmp.com/"], :root a[href^="https://zirdough.net/"], :root [class^="tile-picker__CitrusBannerContainer-sc-"], :root a[href^="https://go.xxxiijmp.com"], :root a[href^="https://go.xtbaffiliates.com/"], :root .OUTBRAIN[data-widget-id^="FMS_REELD_"], :root [data-role="tile-ads-module"], :root a[href^="https://go.xlviirdr.com"], :root a[href^="https://go.xlviiirdr.com"], :root a[href^="https://ismlks.com/"], :root [href^="https://www.mypillow.com/"] > img, :root a[href^="https://go.xlirdr.com"], :root [data-css-class="dfp-inarticle"], :root a[href^="https://l.hyenadata.com/"], :root a[href^="https://go.tmrjmp.com"], :root a[href^="https://go.skinstrip.net"][href*="?campaignId="], :root a[href^="https://go.markets.com/visit/?bta="], :root a[href^="https://billing.purevpn.com/aff.php"] > img, :root a[href^="https://go.hpyrdr.com/"], :root a[href^="https://lijavaxa.com/"], :root a[href^="https://go.goaserv.com/"], :root a[href^="https://t.hrtye.com/"], :root a[href^="https://go.etoro.com/"] > img, :root a[href^="https://go.dmzjmp.com"], :root div[class^="Display_displayAd"], :root a[href^="https://www.bang.com/?aff="], :root #mgb-container > #mgb, :root a[href^="https://go.admjmp.com"], :root div[id^="div-gpt-"], :root a[href^="https://gml-grp.com/"], :root a[href^="https://get.surfshark.net/aff_c?"][href*="&aff_id="] > img, :root a[href^="https://gamingadlt.com/?offer="], :root a[href^="https://www.adskeeper.com"], :root [href^="https://www.reimageplus.com/"], :root a[data-redirect^="https://paid.outbrain.com/network/redir?"], :root [href^="https://clicks.affstrack.com/"] > img, :root a[href^="https://ak.hauchiwu.com/"], :root a[href^="https://engine.phn.doublepimp.com/"], :root a[href^="https://engine.blueistheneworanges.com/"], :root a[href^="https://eergortu.net/"], :root a[href^="https://dl-protect.net/"], :root a[href^="https://disobediencecalculatormaiden.com/"], :root a[href^="https://datingoffers30.info/"], :root a[href*=".foxqck.com/"], :root a[href^="https://ctosrd.com/"], :root a[href^="https://clixtrac.com/"], :root [href^="https://noqreport.com/"] > img, :root a[href^="https://clicks.pipaffiliates.com/"], :root a[href^="https://datewhisper.life/"], :root a[href^="https://get-link.xyz/"], :root a[href^="https://click.linksynergy.com/fs-bin/"] > img, :root a[href^="https://combodef.com/"], :root AD-TRIPLE-BOX, :root a[href^="https://click.hoolig.app/"], :root a[href^="https://track.totalav.com/"], :root img[src^="https://images.purevpnaffiliates.com"], :root a[href^="https://porntubemate.com/"], :root a[href^="https://clickadilla.com/"], :root a[href^="https://click.dtiserv2.com/"], :root a[href^="https://go.xlvirdr.com"], :root a[href^="http://www.iyalc.com/"], :root a[href^="https://claring-loccelkin.com/"], :root a[href^="https://s.ma3ion.com/"], :root a[href^="https://cams.imagetwist.com/in/?track="], :root a[href^="https://t.ajrkm1.com/"], :root [href="https://masstortfinancing.com"] img, :root a[href^="https://bongacams10.com/track?"], :root a[href*=".g2afse.com/"], :root [data-ad-name], :root a[href^="https://bodelen.com/"], :root a[href^="http://revolvemockerycopper.com/"], :root a[href^="https://awptjmp.com/"], :root a[href^="https://join.sexworld3d.com/track/"], :root a[href^="https://intenseaffiliates.com/redirect/"], :root a[href^="https://ads.ad4game.com/"], :root a[href^="https://aweptjmp.com/"], :root [class^="s2nPlayer"], :root a[href^="https://chaturbate.jjgirls.com/?track="], :root a[href^="https://ausoafab.net/"], :root a[href^="https://activate-game.com/"], :root .scroll-fixable.rail-right > .deals-rail, :root [data-wpas-zoneid], :root a[href^="https://a2.adform.net/"], :root a[href^="https://auesk.cfd/"], :root a[href^="https://ak.psaltauw.net/"], :root a[href^="https://adclick.g.doubleclick.net/"], :root div[id^="ad-position-"], :root a[href^="https://leg.xyz/?track="], :root a[href^="https://www.toprevenuegate.com/"], :root a[href^="https://bc.game/"], :root a[href^="https://a.bestcontentoperation.top/"], :root a[href^="https://adultfriendfinder.com/go/"], :root a[href^="https://ads.planetwin365affiliate.com/"], :root a[href^="https://ads.leovegas.com/"], :root [data-m-ad-id], :root a[href^="https://a-ads.com/"], :root div[id^="rc-widget-"], :root a[href^="http://eslp34af.click/"], :root a[href^="https://turnstileunavailablesite.com/"], :root a[href^="https://chaturbate.com/in/?"], :root [href^="https://freecourseweb.com/"] > .sitefriend, :root a[href^="https://prf.hn/click/"][href*="/creativeref:"] > img, :root a[href*="&maxads="], :root a[href^="http://www.adultempire.com/unlimited/promo?"][href*="&partner_id="], :root a[href^="https://1betandgonow.com/"], :root a[href^="https://www.googleadservices.com/pagead/aclk?"], :root a[href^="http://www.mrskin.com/tour"], :root a[href^="http://www.friendlyduck.com/AF_"], :root a[href^="https://getvideoz.click/"], :root a[href^="http://troopsassistedstupidity.com/"], :root a[href^="https://bongacams2.com/track?"], :root amp-ad-custom, :root a[href^="http://trk.globwo.online/"], :root a-ad, :root a[href^="https://offhandpump.com/"], :root a[href^="http://stickingrepute.com/"], :root #slashboxes > .deals-rail, :root a[href^="http://premonitioninventdisagree.com/"], :root a[href^="http://naggingirresponsible.com/"], :root a[href^="https://in.rabbtrk.com/"], :root div[data-ad-targeting], :root a[href^="https://hotplaystime.life/"], :root a[href^="http://www.h4trck.com/"], :root a[href^="https://81ac.xyz/"], :root a[href^="http://guestblackmail.com/"], :root a[href^="http://cam4com.go2cloud.org/aff_c?"], :root a[href^="https://ads.betfair.com/redirect.aspx?"], :root a[href^="https://ad.kubiccomps.icu/"], :root [href^="https://www.mypatriotsupply.com/"] > img, :root a[href^="https://trk.softonixs.xyz/"], :root a[href^="https://sexynearme.com/"], :root a[href^="https://myclick-2.com/"], :root a[href^="http://dragnag.com/"], :root a[style="width:100%;height:100%;z-index:10000000000000000;position:absolute;top:0;left:0;"], :root [id^="div-gpt-ad"], :root .ob_container .item-container-obpd, :root div[id^="yandex_ad"], :root a[href^="https://pb-imc.com/"], :root a[href^="http://deskfrontfreely.com/"], :root a[href^="https://track.adform.net/"], :root a[href^="http://avthelkp.net/"], :root a[href^="https://a.medfoodhome.com/"], :root a[onmousedown^="this.href='http://paid.outbrain.com/network/redir?"][target="_blank"] + .ob_source, :root a[href^="https://engine.flixtrial.com/"], :root [data-type="ad-vertical"], :root a[href^="http://annulmentequitycereals.com/"], :root citrus-ad-wrapper, :root a[href^="https://go.grinsbest.com/"], :root a[onmousedown^="this.href='https://paid.outbrain.com/network/redir?"][target="_blank"], :root a[href^="//startgaming.net/tienda/" i], :root a[href^="https://www.get-express-vpn.com/offer/"], :root a[href^="//s.st1net.com/splash.php"], :root a[href^="https://a.medfoodsafety.com/"], :root a[href^="//go.eabids.com/"], :root a[href^="https://go.nordvpn.net/aff"] > img, :root .\[\&_\.gdprAdTransparencyCogWheelButton\]\:\!pjra-z-\[5\], :root [href^="http://clicks.totemcash.com/"], :root a[href^="https://ad.zanox.com/ppc/"] > img, :root [data-d-ad-id], :root a[href*=".engine.adglare.net/"], :root [href^="https://www.profitablegatecpm.com/"], :root div[id^="lazyad-"], :root a[href^="http://com-1.pro/"], :root a[href*=".cfm?domain="][href*="&fp="], :root [data-dynamic-ads], :root a[href^="https://xbet-4.com/"], :root a[data-widget-outbrain-redirect^="http://paid.outbrain.com/network/redir?"], :root [data-ad-width], :root [data-block-type="ad"], :root [href^="https://join.playboyplus.com/track/"], :root a[data-url^="http://paid.outbrain.com/network/redir?"] + .author, :root a[href^="https://ab.advertiserurl.com/aff/"], :root a[data-oburl^="https://paid.outbrain.com/network/redir?"], :root a[href^="https://trk.sportsflix4k.club/"], :root a[href^="http://dragfault.com/"], :root [href^="https://cpa.10kfreesilver.com/"], :root a[href^="https://a.bestcontentfood.top/"], :root a[data-obtrack^="http://paid.outbrain.com/network/redir?"], :root a[data-href^="http://ads.trafficjunky.net/"], :root a[href^="https://go.xlivrdr.com"], :root [onclick^="location.href='https://1337x.vpnonly.site/"], :root [name^="google_ads_iframe"], :root [id^="section-ad-banner"], :root [data-advadstrackid], :root a[href^="http://muzzlematrix.com/"], :root [href^="https://zone.gotrackier.com/"], :root [href^="https://www.restoro.com/"], :root a[href^="https://bngpt.com/"], :root a[href^="https://www.sheetmusicplus.com/"][href*="?aff_id="], :root a[href^="http://adultfriendfinder.com/go/"], :root a[href^="https://fastestvpn.com/lifetime-special-deal?a_aid="], :root a[href^="https://tour.mrskin.com/"], :root a[href^="https://vo2.qrlsx.com/"], :root [href^="https://www.avantlink.com/click.php"] img, :root a[href^="https://losingoldfry.com/"], :root [href^="https://routewebtk.com/"], :root a[href^="https://t.ajump1.com/"], :root div[id^="div-ads-"], :root [href^="https://rapidgator.net/article/premium/ref/"], :root [href^="https://join.girlsoutwest.com/"], :root [href^="https://join3.bannedsextapes.com"], :root .vid-present > .van_vid_carousel__padding, :root div[id^="adngin-"], :root [data-rc-widget], :root span[data-ez-ph-id], :root [href^="https://track.aftrk1.com/"], :root a[href^="https://go.xxxijmp.com"], :root [href^="https://istlnkcl.com/"], :root div[data-adunit], :root app-large-ad, :root [href^="https://turtlebids.irauctions.com/"] img, :root [href^="https://wct.link/click?"], :root [href^="https://ilovemyfreedoms.com/landing-"], :root [href^="https://go.xlrdr.com"], :root a[href^="https://tm-offers.gamingadult.com/"], :root [href^="https://charmingdatings.life/"], :root [href^="https://glersakr.com/"], :root [href^="https://v.investologic.co.uk/"], :root a[href^="https://www.goldenfrog.com/vyprvpn?offer_id="][href*="&aff_id="], :root [id^="ad-wrap-"], :root a[href*="//jjgirls.com/sex/Chaturbate"], :root [data-id^="div-gpt-ad"], :root a[href^="https://tracker.loropartners.com/"], :root [href^="https://awbbjmp.com/"], :root a[href*=".adsrv.eacdn.com/"], :root [href^="https://antiagingbed.com/discount/"] > img, :root [href^="https://www.brighteonstore.com/products/"] img, :root a[href^="https://bngprm.com/"], :root [href^="https://shiftnetwork.infusionsoft.com/go/"] > img, :root a[href^="https://go.strpjmp.com/"], :root a[href^="https://go.bushheel.com/"], :root a[href^="https://ctjdwm.com/"], :root a[href^="https://camfapr.com/landing/click/"], :root [href="//sexcams.plus/"], :root [href^="http://www.mypillow.com/"] > img, :root [href^="https://www.targetingpartner.com/"], :root div[class^="Adstyled__AdWrapper-"], :root a[href^="https://startgaming.net/tienda/" i], :root .section-subheader > .section-hotel-prices-header, :root [href^="https://www.hostg.xyz/"] > img, :root a[href^="https://join.virtualtaboo.com/track/"], :root [id^="ad_sky"], :root #kt_player > a[target="_blank"], :root a[href^="http://bongacams.com/track?"], :root [href^="http://mypillow.com/"] > img, :root [href="https://ourgoldguy.com/contact/"] img, :root [href="https://www.masstortfinancing.com/"] > img, :root [href="https://jdrucker.com/gold"] > img, :root [href^="https://ad.admitad.com/"], :root [href^="https://mypillow.com/"] > img, :root [data-testid="ad_testID"], :root a[href^="http://handgripvegetationhols.com/"], :root [class^="amp-ad-"], :root a[href^="http://tc.tradetracker.net/"] > img, :root [data-ez-name], :root a[href^="https://black77854.com/"], :root [data-revive-zoneid] > iframe, :root [data-testid="adBanner-wrapper"], :root [href^="https://mylead.global/stl/"] > img, :root [href^="https://mypatriotsupply.com/"] > img, :root a[href^="https://go.hpyjmp.com"], :root [href^="https://mystore.com/"] > img, :root a[href^="https://witnessjacket.com/"], :root [data-mobile-ad-id], :root a[href^="https://fc.lc/ref/"], :root [data-adshim], :root topadblock, :root a[href^="//s.zlinkd.com/"], :root a[href^="https://bs.serving-sys.com"], :root [href^="https://goldcometals.com/clk.trk"], :root [data-desktop-ad-id], :root [href^="https://www.cloudways.com/en/?id"], :root a[href^="https://www.liquidfire.mobi/"], :root .grid > .container > #aside-promotion, :root DFP-AD, :root [id^="ad_slider"], :root [data-adbridg-ad-class], :root a[href^="https://linkboss.shop/"], :root [data-adblockkey], :root a[href^="https://go.cmtaffiliates.com/"], :root [href^="https://optimizedelite.com/"] > img, :root [data-name="adaptiveConstructorAd"], :root [onclick*="content.ad/"], :root [data-ad-manager-id], :root AMP-AD, :root [data-ad-cls], :root a[href^="https://6-partner.com/"], :root [href^="https://affiliate.fastcomet.com/"] > img, :root a[href^="https://go.mnaspm.com/"], :root [class^="adDisplay-module"], :root AD-SLOT, :root #kt_player > div[style="position: absolute; inset: 0px; z-index: 170;"], :root [data-freestar-ad][id], :root a[href^="https://go.xxxjmp.com"], :root #leader-companion > a[href], :root a[href^="http://www.adultdvdempire.com/?partner_id="][href*="&utm_"], :root [data-ad-module], :root a[href^="http://partners.etoro.com/"], :root a[href^=" https://www.friendlyduck.com/AF_"], :root a[href*="/jump/next.php?r="], :root app-advertisement, :root a[href^="https://getmatchedlocally.com/"], :root a[href^="https://clickins.slixa.com/"], :root [class^="div-gpt-ad"], :root a[href^="https://traffdaq.com/"], :root a[href^="https://cam4com.go2cloud.org/"], :root a[href^="http://li.blogtrottr.com/click?"], :root .ob_dual_right > .ob_ads_header ~ .odb_div { display: none !important; }</style></head>
  <body><div class="menu_container"><div class="menu_info blurb" id="menu_info"></div><a class="menu_header" href="https://meme-suite.org/meme/index.html">MEME Suite 5.5.5</a><div class="menu_topic_jobqueue" data-info="Status of job queue. Click to update." data-id="0">Jobs running: 14<br>Jobs waiting to run: 31</div><div class="menu_subtopics" style="max-height: 0px;"><div></div></div><div class="menu_topic open" data-info="&lt;i&gt;De novo&lt;/i&gt; discovery of motifs in set(s) of sequences." data-id="1"><div class="menu_arrow closed">▶</div><div class="menu_arrow open">▼</div>Motif Discovery</div><div class="menu_subtopics" style="max-height: 136px;"><div><a href="https://meme-suite.org/meme/tools/meme" class="menu_subtopic" data-info="&lt;dfn&gt;MEME&lt;/dfn&gt; discovers novel, &lt;em&gt;ungapped&lt;/em&gt; motifs (recurring, fixed-length patterns) in your sequences&lt;span class=&quot;sample_output&quot;&gt; (&lt;a href=&quot;../doc/examples/meme_example_output_files/meme.html&quot;&gt;sample output&lt;/a&gt; from &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/crp0.fna&#39;&gt;sequences&lt;/a&gt;)&lt;/span&gt;. MEME splits variable-length patterns into two or more separate motifs. &lt;span class=&quot;manual_link&quot;&gt;See this &lt;a href=&quot;../doc/meme.html?man_type=web&quot;&gt;Manual&lt;/a&gt; for more information.&lt;/span&gt;  &lt;footer&gt;&lt;br&gt;&lt;br&gt;A motif is an approximate sequence pattern that occurs repeatedly in a group of related sequences. MEME represents motifs as position-dependent letter-probability matrices that describe the probability of each possible letter at each position in the pattern.  Individual MEME motifs do not contain gaps. Patterns with variable-length gaps are split by MEME into two or more separate motifs. &lt;br&gt;&lt;br&gt;MEME takes as input a group of sequences and outputs as many motifs as requested. MEME uses statistical modeling techniques to automatically choose the best width, number of occurrences, and description for each motif. &lt;br&gt;&lt;br&gt;MEME on the web can take a second (control) set of input sequences and then discovers motifs that are enriched in the primary set relative to the control set.  This is called discriminative motif discovery. &lt;br&gt;&lt;br&gt;&lt;b&gt;Note:&lt;/b&gt; If you have more than 50 sequences, you might want to use STREME instead of MEME. &lt;br&gt;&lt;br&gt;&lt;b&gt;Note:&lt;/b&gt; MEME treats all ambiguous characters in sequences as wildcards that match every letter equally well.&lt;/footer&gt;">MEME</a><a href="https://meme-suite.org/meme/tools/streme" class="menu_subtopic" data-info="&lt;dfn&gt;STREME&lt;/dfn&gt; discovers &lt;em&gt;ungapped&lt;/em&gt; motifs (recurring, fixed-length patterns) that are &lt;em&gt;enriched&lt;/em&gt; in your sequences or &lt;em&gt;relatively enriched&lt;/em&gt; in them compared to your control sequences&lt;span class=&quot;sample_output&quot;&gt; (&lt;a href=&quot;../doc/examples/streme_example_output_files/streme.html&quot;&gt;sample output&lt;/a&gt; from &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/Klf1.fna&#39;&gt;sequences&lt;/a&gt;)&lt;/span&gt;.&lt;span class=&quot;manual_link&quot;&gt; See this &lt;a href=&quot;../doc/streme.html?man_type=web&quot;&gt;Manual&lt;/a&gt; or this &lt;a href=&quot;../doc/streme-tutorial.html&quot;&gt;Tutorial&lt;/a&gt; for more information.&lt;/span&gt;  &lt;footer&gt;&lt;br&gt;&lt;br&gt;STREME (Simple, Thorough, Rapid, Enriched Motif Elicitation) finds motifs (up to 30 columns wide) in large sequence datasets. The input to STREME is one or two sets of sequences.  The control sequences should have approximately the same length distribution as the primary sequences. If you do not provide a control set, the program shuffles the primary set to create a control set. The program uses Fisher&#39;s Exact Test or the Binomial test to determine significance of each motif found in the positive set as compared with its representation in the control set, using a significance threshold that may be set on the command line.&lt;br&gt;&lt;br&gt;STREME achieves high speed via an optimization algorithm that uses a suffix tree, and provides &lt;b&gt;accurate estimates&lt;/b&gt; of the statistical significance of the motifs it discovers. &lt;br&gt;&lt;br&gt;&lt;b&gt;Note:&lt;/b&gt; If you have fewer than 50 sequences, you might want to use MEME instead of STREME. &lt;br&gt;&lt;br&gt;&lt;b&gt;Note:&lt;/b&gt; STREME will not include sequence positions that contain ambiguous characters in the motifs it discovers, and will work with DNA sequences even if you specify the RNA alphabet.&lt;/footer&gt;">STREME</a><a href="https://meme-suite.org/meme/tools/xstreme" class="menu_subtopic" data-info="&lt;dfn&gt;XSTREME&lt;/dfn&gt; performs &lt;em&gt;comprehensive motif analysis&lt;/em&gt; (including motif discovery) on sequences where the motif sites can be &lt;em&gt;anywhere&lt;/em&gt; in the sequences&lt;span class=&quot;sample_output&quot;&gt; (&lt;a href=&quot;../doc/examples/xstreme_example_output_files/xstreme.html&quot;&gt;sample output&lt;/a&gt; from &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/Klf1.fna&#39;&gt;sequences&lt;/a&gt;)&lt;/span&gt;. The input sequences may be of &lt;em&gt;any length&lt;/em&gt;, and their lengths may &lt;em&gt;vary&lt;/em&gt;. &lt;span class=&quot;manual_link&quot;&gt;See this &lt;a href=&quot;../doc/xstreme.html?man_type=web&quot;&gt;Manual&lt;/a&gt; for more information.&lt;/span&gt;  &lt;footer&gt;&lt;br&gt;&lt;br&gt;XSTREME will: &lt;ol&gt; &lt;li&gt;Discover novel motifs in the input sequences (with STREME and MEME).&lt;/li&gt; &lt;li&gt;Determine which motifs are most enriched (with SEA).&lt;/li&gt; &lt;li&gt;Analyze them for similarity to known motifs (with Tomtom).&lt;/li&gt; &lt;li&gt;Group significant motifs by similarity.&lt;/li&gt; &lt;/li&gt;&lt;li&gt;Create a GFF file for viewing each motif&#39;s predicted sites in a genome browser.&lt;/li&gt;&lt;/ol&gt;It is worth noting that XSTREME is &lt;b&gt;not&lt;/b&gt; a motif scanner, but the motifs discovered by it can be used by FIMO, MAST or MCAST to scan for motif sites.&lt;/footer&gt;">XSTREME</a><a href="https://meme-suite.org/meme/tools/meme-chip" class="menu_subtopic" data-info="&lt;dfn&gt;MEME-ChIP&lt;/dfn&gt; performs &lt;em&gt;comprehensive motif analysis&lt;/em&gt; (including motif discovery) on sequences where the motif sites tend to be &lt;em&gt;centrally&lt;/em&gt; located, such as ChIP-seq peaks&lt;span class=&quot;sample_output&quot;&gt; (&lt;a href=&quot;../doc/examples/memechip_example_output_files/meme-chip.html&quot;&gt;sample output&lt;/a&gt; from &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/Klf1.fna&#39;&gt;sequences&lt;/a&gt;)&lt;/span&gt;. The input sequences should be &lt;em&gt;centered&lt;/em&gt; on a &lt;em&gt;100 character region&lt;/em&gt; expected to contain motifs, and each sequence should ideally be around &lt;em&gt;500 letters&lt;/em&gt; long. &lt;/em&gt;&lt;span class=&quot;manual_link&quot;&gt; See this &lt;a href=&quot;../doc/meme-chip.html?man_type=web&quot;&gt;Manual&lt;/a&gt; for more information.&lt;/span&gt;  &lt;footer&gt;&lt;br&gt;&lt;br&gt;MEME-ChIP will: &lt;ol&gt; &lt;li&gt;Discover novel motifs in the &lt;b&gt;central regions&lt;/b&gt; (100 characters by default) of the input sequences (with MEME and STREME).&lt;/li&gt;&lt;li&gt;Determine which motifs are most centrally enriched (with CentriMo).&lt;/li&gt;&lt;li&gt;Analyze them for similarity to known motifs (with Tomtom).&lt;/li&gt;&lt;li&gt;Group significant motifs by similarity.&lt;/li&gt;&lt;li&gt;Perform a motif spacing analysis (with SpaMo).&lt;/li&gt;&lt;li&gt;Create a GFF file for viewing each motif&#39;s predicted sites in a genome browser.&lt;/li&gt;&lt;/ol&gt;&lt;/footer&gt;">MEME-ChIP</a><a href="https://meme-suite.org/meme/tools/glam2" class="menu_subtopic" data-info="&lt;dfn&gt;GLAM2&lt;/dfn&gt; discovers novel, &lt;em&gt;gapped&lt;/em&gt; motifs (recurring, variable-length patterns) in your &lt;em&gt;DNA&lt;/em&gt; or &lt;em&gt;protein&lt;/em&gt; sequences&lt;span class=&quot;sample_output&quot;&gt; (&lt;a href=&quot;../doc/examples/glam2_example_output_files/glam2.html&quot;&gt;sample output&lt;/a&gt; from &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/At.faa&#39;&gt;sequences&lt;/a&gt;)&lt;/span&gt;.&lt;span class=&quot;manual_link&quot;&gt; See this &lt;a href=&quot;../doc/glam2.html?man_type=web&quot;&gt;Manual&lt;/a&gt; for more information.&lt;/span&gt;  &lt;footer&gt;&lt;br&gt;&lt;br&gt;GLAM2 is a program for finding motifs in sequences, typically amino-acid or nucleotide sequences. The main innovation of GLAM2 is that it allows insertions and deletions in motifs.  &lt;br&gt;&lt;br&gt;GLAM2 is a child of &lt;a href=&quot;http://zlab.bu.edu/glam/&quot;&gt;GLAM&lt;/a&gt; and a sibling of &lt;a href=&quot;http://www.ncbi.nlm.nih.gov/CBBresearch/Spouge/html.ncbi/index/software.html#7&quot;&gt;A-GLAM&lt;/a&gt;.  &#39;GLAM&#39; previously to stood for &#39;Gapless Local Alignment of Multiple sequences&#39;, but it now stands for &#39;Gapped Local Alignment of Motifs&#39;.  &lt;br&gt;&lt;br&gt;GLAM2 uses a &lt;a href=&quot;http://en.wikipedia.org/wiki/Simulated_annealing&quot;&gt;simulated annealing&lt;/a&gt; algorithm, with a temperature parameter. At high temperatures, GLAM2 only slightly favors changes that increase the alignment&#39;s score, and at low temperatures it strongly favors such changes. Thus, at high temperatures the score will be optimized too slowly, but at low temperatures the algorithm will get frozen in a local optimum. The strategy, then, is to start with a high temperature and reduce it as slowly as possible.  &lt;br&gt;&lt;br&gt;Refer to the &lt;a href=&quot;glam2_tut.html&quot;&gt;GLAM2 tutorial&lt;/a&gt;, which uses the command-line version of GLAM2, if you need more information.&lt;/footer&gt;">GLAM2</a><a href="https://meme-suite.org/meme/tools/momo" class="menu_subtopic" data-info="&lt;dfn&gt;MoMo&lt;/dfn&gt; discovers sequence motifs associated with different types of protein post-translational modifications (PTMs)&lt;span class=&quot;sample_output&quot;&gt; (&lt;a href=&quot;../doc/examples/momo_example_output_files/momo.html&quot;&gt;sample output&lt;/a&gt;)&lt;/span&gt;. The program takes as input a collection of PTMs identified using protein mass spectrometry.  For each distinct type of PTM, MoMo uses one of three algorithms to discover motifs representing amino acid preferences flanking the modification site. &lt;span class=&quot;manual_link&quot;&gt;See this &lt;a href=&quot;../doc/momo.html?man_type=web&quot;&gt;Manual&lt;/a&gt; for more information.&lt;/span&gt;  &lt;footer&gt;&lt;br&gt;&lt;br&gt;&lt;b&gt;Note:&lt;/b&gt; MoMo treats all ambiguous characters in sequences as wildcards that match every letter equally well.&lt;/footer&gt;">MoMo</a><a href="https://meme-suite.org/meme/tools/dreme" class="menu_subtopic" data-info="&lt;dfn&gt;DREME (deprecated; please consider using &lt;em&gt;STREME&lt;/em&gt; instead)&lt;/dfn&gt; discovers &lt;em&gt;short&lt;/em&gt;, &lt;em&gt;ungapped&lt;/em&gt; motifs (recurring, fixed-length patterns) that are &lt;em&gt;relatively&lt;/em&gt; enriched in your sequences compared with shuffled sequences or your control sequences&lt;span class=&quot;sample_output&quot;&gt; (&lt;a href=&quot;../doc/examples/dreme_example_output_files/dreme.html&quot;&gt;sample output&lt;/a&gt; from &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/Klf1.fna&#39;&gt;sequences&lt;/a&gt;)&lt;/span&gt;.&lt;span class=&quot;manual_link&quot;&gt; See this &lt;a href=&quot;../doc/dreme.html?man_type=web&quot;&gt;Manual&lt;/a&gt; or this &lt;a href=&quot;../doc/dreme-tutorial.html&quot;&gt;Tutorial&lt;/a&gt; for more information.&lt;/span&gt;  &lt;footer&gt;&lt;br&gt;&lt;br&gt;DREME (Discriminative Regular Expression Motif Elicitation) finds relatively short motifs (up to 8 positions) fast. The input to DREME is one or two sets of sequences.  The control sequences should be approximately the same length as the primary sequences. If you do not provide a control set, the program shuffles the primary set to provide a control set. The program uses Fisher&#39;s Exact Test to determine significance of each motif found in the positive set as compared with its representation in the control set, using a significance threshold that may be set on the command line.&lt;br&gt;&lt;br&gt;DREME achieves its high speed by restricting its search to regular expressions based on the symbols available in the alphabet, and by using a heuristic estimate of generalized motifs&#39; statistical significance.  &lt;br&gt;&lt;br&gt;&lt;b&gt;Note:&lt;/b&gt; DREME ignores sequence positions that contain ambiguous characters.&lt;/footer&gt;">DREME (deprecated)</a></div></div><div class="menu_topic" data-info="Analyze sequence set(s) for enrichment of known motifs or motifs you provide." data-id="2"><div class="menu_arrow closed">▶</div><div class="menu_arrow open">▼</div>Motif Enrichment</div><div class="menu_subtopics" style="max-height: 0px;"><div><a href="https://meme-suite.org/meme/tools/sea" class="menu_subtopic" data-info="&lt;dfn&gt;SEA&lt;/dfn&gt; identifies &lt;em&gt;known&lt;/em&gt; or &lt;em&gt;user-provided&lt;/em&gt; motifs that are &lt;em&gt;relatively&lt;/em&gt; enriched in your sequences compared with shuffled sequences or your control sequences&lt;span class=&quot;sample_output&quot;&gt; (&lt;a href=&quot;../doc/examples/sea_example_output_files/sea.html&quot;&gt;sample output&lt;/a&gt; from &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/Klf1.fna&#39;&gt;sequences&lt;/a&gt;)&lt;/span&gt;.&lt;span class=&quot;manual_link&quot;&gt; See this &lt;a href=&quot;../doc/sea.html?man_type=web&quot;&gt;Manual&lt;/a&gt; for more information.&lt;/span&gt;  &lt;footer&gt;&lt;br&gt;&lt;br&gt;SEA (Simple Enrichment Analysis) applies the same objective function used by the STREME motif discovery algorithm to measure the enrichment of motifs. The input to SEA is one or two sets of sequences.  If you provide a control set of sequences, they should have approximately the same length distribution as the primary sequences. If you do not provide a control set, SEA shuffles each sequence in the primary set to create a control set. Enrichment is based on the best match to the motif in each sequence (on either strand when the alphabet is complementable).&lt;br&gt;&lt;br&gt;&lt;b&gt;Note:&lt;/b&gt; SEA ignores sequence positions that contain ambiguous characters, and will work with DNA sequences even if the motifs use the RNA alphabet.&lt;/footer&gt;">SEA</a><a href="https://meme-suite.org/meme/tools/centrimo" class="menu_subtopic" data-info="&lt;dfn&gt;CentriMo&lt;/dfn&gt; identifies &lt;em&gt;known&lt;/em&gt; or &lt;em&gt;user-provided&lt;/em&gt; motifs that show a significant preference for particular &lt;em&gt;locations&lt;/em&gt; in your sequences&lt;span class=&quot;sample_output&quot;&gt; (&lt;a href=&quot;../doc/examples/centrimo_example_output_files/centrimo.html&quot;&gt;sample output&lt;/a&gt; from &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/mm9_tss_500bp_sampled_1000.fna&#39;&gt;sequences&lt;/a&gt; and &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/some_vertebrates.meme&#39;&gt;motifs&lt;/a&gt;)&lt;/span&gt;. CentriMo can also show if the local enrichment is significant &lt;em&gt;relative&lt;/em&gt; to control sequences. &lt;span class=&quot;manual_link&quot;&gt;See this &lt;a href=&quot;../doc/centrimo.html?man_type=web&quot;&gt;Manual&lt;/a&gt; or this &lt;a href=&quot;../doc/centrimo-tutorial.html&quot;&gt;Tutorial&lt;/a&gt; for more information.&lt;/span&gt; &lt;footer&gt;&lt;br&gt;&lt;br&gt;CentriMo takes a set of motifs and a set of equal-length sequences and plots the positional distribution of the best match of each motif.  The motifs are typically compendia of DNA- or RNA-binding motifs, and the sequences might be: 500 bp sequences aligned on ChIP-seq peaks or summits; 300 bp sequences centered on sets of transcription start sites or translation start sites; sequences aligned on splice-junctions; etc.  The web version of CentriMo provides a large number of motif databases, or you can upload your own motifs in &lt;a href=meme-format.html&gt;MEME Motif Format&lt;/a&gt;, or enter them by typing in many additional formats. &lt;br&gt;&lt;br&gt;CentriMo also computes the &#39;local enrichment&#39; of each motif by counting the number of times its best match in each sequence occurs in a local region and applying a statistical test to see if the local enrichment is significant.  By default, CentriMo examines only regions &lt;b&gt;centered&lt;/b&gt; in the input sequences, but CentriMo will compute the enrichment of &lt;b&gt;all&lt;/b&gt; regions if requested.  CentriMo uses the binomial test to compute the significance of the number of sequences where the best match occurs in a given region, assuming a uniform prior over best match positions.  CentriMo reports the location and significance of the best region for each motif.  &lt;br&gt;&lt;br&gt; CentriMo can also perform comparative enrichment, reporting the relative enrichment of the best region in a second, control set of sequences if you provide a set of control sequences.  CentriMo chooses regions based on their significance in the primary set of sequences, and then it uses Fisher&#39;s exact test to evaluate the significance of the number of best matches in the region in the primary set compared with the number of best matches in the same region in the control set of sequences.&lt;br&gt;&lt;br&gt;&lt;b&gt;Note:&lt;/b&gt; CentriMo does not score sequence positions that contain ambiguous characters.&lt;/footer&gt;">CentriMo</a><a href="https://meme-suite.org/meme/tools/ame" class="menu_subtopic" data-info="&lt;dfn&gt;AME&lt;/dfn&gt; identifies &lt;em&gt;known&lt;/em&gt; or &lt;em&gt;user-provided&lt;/em&gt; motifs that are either &lt;em&gt;relatively&lt;/em&gt; enriched in your sequences compared with control sequences, that are enriched in the first sequences in your input file, or that are enriched in sequences with &lt;b&gt;small&lt;/b&gt; values of scores that you can specify with your input sequences &lt;span class=&quot;sample_output&quot;&gt; (&lt;a href=&quot;../doc/examples/ame_example_output_files/ame.html&quot;&gt;sample output&lt;/a&gt; from &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/crp0.fna&#39;&gt;sequences&lt;/a&gt;, &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/lex0.fna&#39;&gt;control sequences&lt;/a&gt; and &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/dpinteract.meme&#39;&gt;motifs&lt;/a&gt;)&lt;/span&gt;. &lt;span class=&quot;manual_link&quot;&gt;See this &lt;a href=&quot;../doc/ame.html?man_type=web&quot;&gt;Manual&lt;/a&gt; or this &lt;a href=&quot;../doc/ame-tutorial.html&quot;&gt;Tutorial&lt;/a&gt; for more information.&lt;/span&gt; &lt;footer&gt;&lt;br&gt;&lt;br&gt;AME (Analysis of Motif Enrichment) scores a set of sequences with a motif, treating each subsequence (and its reverse complement for complementable alphabets) in the sequence as a possible match to the motif. AME supports several types of sequence scoring functions, and it treats motif occurrences the same, regardless of their locations within the sequences.  AME supports several types of statistical enrichment functions.  (See the Command-line Version of the &lt;a href=&quot;../doc/ame.html?man_type=web&quot;&gt;Manual&lt;/a&gt; for more details.) The web version of AME provides a large number of motif databases, or you can upload your own motifs in &lt;a href=meme-format.html&gt;MEME Motif Format&lt;/a&gt;, or enter them by typing in many additional formats.&lt;br&gt;&lt;br&gt;&lt;b&gt;Note:&lt;/b&gt; AME does not score sequence positions that contain ambiguous characters.&lt;/footer&gt;">AME</a><a href="https://meme-suite.org/meme/tools/spamo" class="menu_subtopic" data-info="&lt;dfn&gt;SpaMo&lt;/dfn&gt; identifies significantly enriched &lt;em&gt;spacings&lt;/em&gt; in a set of sequences between a &lt;em&gt;primary motif&lt;/em&gt; and each motif in a set of &lt;em&gt;secondary motifs&lt;/em&gt;&lt;span class=&quot;sample_output&quot;&gt; (&lt;a href=&quot;../doc/examples/spamo_example_output_files/spamo.html&quot;&gt;sample output&lt;/a&gt; from &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/Klf1.fna&#39;&gt;sequences&lt;/a&gt; and &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/Klf1.meme&#39;&gt;motif&lt;/a&gt;)&lt;/span&gt;. Typically, the input sequences are centered on ChIP-seq peaks, and are each 500bp long.  &lt;span class=&quot;manual_link&quot;&gt;See this &lt;a href=&quot;../doc/spamo.html?man_type=web&quot;&gt;Manual&lt;/a&gt; for more information.&lt;/span&gt;  &lt;footer&gt;&lt;br&gt;&lt;br&gt;The name SpaMo stands for &#39;Spaced Motif&#39; analysis.  Its inputs are a set of many short sequences, a primary motif and one or more databases of secondary motifs. It searches for the strongest primary motif site (on both strands for complementable alphabets) and then searches in the area around it looking for the strongest secondary motif site. The relative spacings of the primary and secondary motif in all the sequences is tallied and the probability of the close spacings happening by chance is calculated.  &lt;br&gt;&lt;br&gt; After all the calculations are done SpaMo, outputs the non-redundant secondary motifs in order of significance provided they had a bin that passed the significance threshold. Similar secondary motifs are grouped together and listed in order of significance on the secondary motif they were redundant to. If the bin size is 1 (as it is with the web version of SpaMo), then an alignment for each of the similar secondary motifs is created.  &lt;br&gt;&lt;br&gt;The web version of SpaMo provides a large number of secondary motif databases, or you can upload your own motifs in &lt;a href=meme-format.html&gt;MEME Motif Format&lt;/a&gt;, or enter them by typing in many additional formats. &lt;br&gt;&lt;br&gt;It is important to note that the locations searched for the primary and secondary motifs are dependent on a SpaMO parameter called the &quot;margin&quot;, which is 150bp by default. The secondary motif must be contained within an area limited by the margin size on either side of the strongest primary motif site. Additionally the primary motif cannot be found closer than the margin size to the edge of the input sequence.&lt;b&gt;&lt;br&gt;&lt;figure&gt; &lt;figcaption&gt;Restrictions on primary motif locations.&lt;/figcaption&gt; &lt;canvas id=primary_site_diagram&gt;&lt;/canvas&gt; &lt;/figure&gt; &lt;figure&gt; &lt;figcaption&gt;Restrictions on secondary motif locations.&lt;/figcaption&gt; &lt;canvas id=secondary_site_diagram&gt;&lt;/canvas&gt; &lt;/figure&gt; &lt;b&gt;Note:&lt;/b&gt; SpaMo does not score sequence positions that contain ambiguous characters.&lt;/footer&gt;">SpaMo</a><a href="https://meme-suite.org/meme/tools/gomo" class="menu_subtopic" data-info="&lt;dfn&gt;GOMo&lt;/dfn&gt; scans all &lt;em&gt;promoters&lt;/em&gt; using &lt;em&gt;nucleotide motifs&lt;/em&gt; you provide to determine if any motif is significantly associated with genes linked to one or more Genome Ontology (GO) terms&lt;span class=&quot;sample_output&quot;&gt; (&lt;a href=&quot;../doc/examples/gomo_example_output_files/gomo.html&quot;&gt;sample output&lt;/a&gt; from &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/dpinteract_subset.meme&#39;&gt;motifs&lt;/a&gt; and E.coli K12 database)&lt;/span&gt;. The significant GO terms can suggest the &lt;em&gt;biological roles&lt;/em&gt; of the motifs.  &lt;span class=&quot;manual_link&quot;&gt;See this &lt;a href=&quot;../doc/gomo.html?man_type=web&quot;&gt;Manual&lt;/a&gt; for more information.&lt;/span&gt;  &lt;footer&gt;&lt;br&gt;&lt;br&gt;The name GOMo stands for &#39;Gene Ontology for Motifs&#39;.  The program searches in a set of ranked genes for enriched GO terms associated with high-ranking genes. The genes can be ranked, for example, by applying a motif scoring algorithms on their upstream sequence.  &lt;br&gt;&lt;br&gt;GOMo computes a score for each GO term by estimating the Mann-Whitney rank-sum &lt;i&gt;p&lt;/i&gt;-value of the GO term&#39;s genes, and can combine the &lt;i&gt;p&lt;/i&gt;-values for multiple species. It then estimates final &lt;i&gt;p&lt;/i&gt;-values for each GO-term empirically by shuffling the gene identifiers in the ranking (ensuring consistency across species) to generate scores from the null hypothesis. Then &lt;i&gt;q&lt;/i&gt;-values are derived from these &lt;i&gt;p&lt;/i&gt;-values following the method of Benjamini and Hochberg (where &#39;&lt;i&gt;q&lt;/i&gt;-value&#39; is defined as the minimal false discovery rate at which a given GO-term is deemed significant).&lt;br&gt;&lt;br&gt;The program reports all GO terms that receive &lt;i&gt;q&lt;/i&gt;-values smaller than a specified threshold, outputting a GOMo score with empirically calculated &lt;i&gt;p&lt;/i&gt;-values and &lt;i&gt;q&lt;/i&gt;-values for each.&lt;/footer&gt;">GOMo</a></div></div><div class="menu_topic" data-info="Find matches to motif(s) in sequences." data-id="3"><div class="menu_arrow closed">▶</div><div class="menu_arrow open">▼</div>Motif Scanning</div><div class="menu_subtopics" style="max-height: 0px;"><div><a href="https://meme-suite.org/meme/tools/fimo" class="menu_subtopic" data-info="&lt;dfn&gt;FIMO&lt;/dfn&gt; scans a set of sequences for &lt;em&gt;individual matches&lt;/em&gt; to each of the motifs you provide&lt;span class=&quot;sample_output&quot;&gt; (&lt;a href=&quot;../doc/examples/fimo_example_output_files/fimo.html&quot;&gt;sample output&lt;/a&gt; for &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/some_vertebrates.meme&#39;&gt;motifs&lt;/a&gt; and &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/mm9_tss_500bp_sampled_1000.fna&#39;&gt;sequences&lt;/a&gt;)&lt;/span&gt;.&lt;span class=&quot;manual_link&quot;&gt; See this &lt;a href=&quot;../doc/fimo.html?man_type=web&quot;&gt;Manual&lt;/a&gt; or this &lt;a href=&quot;../doc/fimo-tutorial.html&quot;&gt;Tutorial&lt;/a&gt; for more information.&lt;/span&gt;  &lt;footer&gt;&lt;br&gt;&lt;br&gt;The name FIMO stands for &#39;Find Individual Motif Occurrences&#39;.  The program searches a set of sequences for occurrences of known motifs, treating each motif independently.  Motifs must be in &lt;a href=meme-format.html&gt;MEME Motif Format&lt;/a&gt;.  The web version of FIMO also allows you to type in motifs in additional formats. &lt;br&gt;&lt;br&gt;&lt;b&gt;Note:&lt;/b&gt; FIMO does not score sequence positions that contain ambiguous characters.&lt;/footer&gt;">FIMO</a><a href="https://meme-suite.org/meme/tools/mast" class="menu_subtopic" data-info="&lt;dfn&gt;MAST&lt;/dfn&gt; searches sequences for matches to a set of motifs, and sorts the sequences by the &lt;em&gt;best combined match&lt;/em&gt; to all motifs&lt;span class=&quot;sample_output&quot;&gt; (&lt;a href=&quot;../doc/examples/mast_example_output_files/mast.html&quot;&gt;sample output&lt;/a&gt; for &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/adh.meme&#39;&gt;motifs&lt;/a&gt; and &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/adh.faa&#39;&gt;sequences&lt;/a&gt;).&lt;/span&gt;&lt;span class=&#39;manual_link&#39;&gt; See this &lt;a href=&quot;../doc/mast.html?man_type=web&quot;&gt;Manual&lt;/a&gt; for more information.&lt;/span&gt;  &lt;footer&gt;&lt;br&gt;&lt;br&gt;The name MAST stands for &#39;Motif Alignment and Search Tool&#39;.  A motif is a sequence pattern that occurs repeatedly in a group of related sequences. Motifs are represented as position-dependent probability (or scoring) matrices that describe the probability (or score) of each possible letter at each position in the pattern. Individual motifs may not contain gaps. Patterns with variable-length gaps must be split into two or more separate motifs before being submitted as input to MAST.  &lt;br&gt;&lt;br&gt;MAST takes as input a file containing the descriptions of one or more motifs and searches a set of sequences that you select for sequences that match the motifs.  Motifs must be in &lt;a href=meme-format.html&gt;MEME Motif Format&lt;/a&gt;. The web version of MAST also allows you to type in motifs in additional formats.  &lt;br&gt;&lt;br&gt;MAST can search sequences with motifs of the same alphabet but as a special feature it can also search protein sequences with DNA motifs.  You can control how (or if) MAST scans the reverse complement strand (if applicable).  You can also set thresholds for overall (combined match) significance and individual motif site significance.&lt;br&gt;&lt;br&gt;&lt;b&gt;Note:&lt;/b&gt; MAST scores sequence positions that contain ambiguous characters using the weighted average of the log-odds scores of the letters that the ambiguous character matches. The weights are the background frequencies of those letters.&lt;/footer&gt;">MAST</a><a href="https://meme-suite.org/meme/tools/mcast" class="menu_subtopic" data-info="&lt;dfn&gt;MCAST&lt;/dfn&gt; searches sequences for &lt;em&gt;clusters of matches&lt;/em&gt; to one or more &lt;em&gt;nucleotide&lt;/em&gt; motifs&lt;span class=&quot;sample_output&quot;&gt; (&lt;a href=&quot;../doc/examples/mcast_example_output_files/mcast.html&quot;&gt;sample output&lt;/a&gt; for &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/crp0.meme&#39;&gt;motifs&lt;/a&gt; and &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/crp0.fna&#39;&gt;sequences&lt;/a&gt;)&lt;/span&gt;.&lt;span class=&#39;manual_link&#39;&gt; See this &lt;a href=&quot;../doc/mcast.html?man_type=web&quot;&gt;Manual&lt;/a&gt; for more information.&lt;/span&gt;  &lt;footer&gt;&lt;br&gt;&lt;br&gt;The name MCAST stands for &#39;Motif Cluster Alignment Search Tool&#39;.  MCAST searches a set of sequences for statistically significant clusters of &lt;b&gt;non-overlapping&lt;/b&gt; occurrences of a given set of motifs.  It is primarily designed for search genomic DNA databases. Motifs must be in &lt;a href=meme-format.html&gt;MEME Motif Format&lt;/a&gt;. The web version of MCAST also allows you to type in motifs in additional formats.  &lt;br&gt;&lt;br&gt;A motif &#39;hit&#39; is a sequence position that is sufficiently similar to a motif in the query, where the score for a motif at a particular sequence position is computed without gaps. To compute the &lt;i&gt;p-&lt;/i&gt;value of a motif score, MCAST assumes that the sequences in the file were generated by a 0-order Markov process.  To be considered a hit, the &lt;i&gt;p&lt;/i&gt;-value of the motif alignment score must be less than the significance threshold, which you can specify.  Note that MCAST searches for hits on both strands of the sequences.  &lt;br&gt;&lt;br&gt;A cluster of non-overlapping hits is called a &#39;match&#39;. You can specify the maximum allowed distance between the hits in a match.  Two hits separated by more than the maximum allowed gap will be reported in separate matches.  &lt;br&gt;&lt;br&gt;The &lt;i&gt;p-&lt;/i&gt;value of a hit is converted to a &#39;p-score&#39; in order to compute the total score of the match it participates in. The p-score for a hit with &lt;i&gt;p-&lt;/i&gt;value &lt;span class=&quot;pdat&quot;&gt;p&lt;/span&gt; is &lt;div class=&quot;indent&quot;&gt; S = -log&lt;sub&gt;2&lt;/sub&gt;(&lt;span class=&quot;pdat&quot;&gt;p&lt;/span&gt;/&lt;span class=&quot;pdat&quot;&gt;pthresh&lt;/span&gt;), &lt;/div&gt; &lt;p style=&quot;margin-top:0&quot;&gt; where &lt;span class=&quot;pdat&quot;&gt;pthresh&lt;/span&gt; is 0.0005 by default.  The total score of a match is the sum of the p-scores of the hits making up the match.&lt;/p&gt; &lt;p style=&quot;margin-bottom:0&quot;&gt;MCAST searches for all possible matches between the query motifs and the sequences, and reports the matches with the largest scores in decreasing order. Three types of statistical confidence estimates (&lt;i&gt;p&lt;/i&gt;-value, &lt;i&gt;E&lt;/i&gt;-value, and &lt;i&gt;q&lt;/i&gt;-value) are estimated for each score, and the reported matches can be filtered by applying &lt;i&gt;p&lt;/i&gt;-value or &lt;i&gt;q&lt;/i&gt;-value thresholds that you can specify.&lt;/footer&gt;">MCAST</a><a href="https://meme-suite.org/meme/tools/glam2scan" class="menu_subtopic" data-info="&lt;dfn&gt;GLAM2Scan&lt;/dfn&gt; searches sequences for matches to gapped &lt;em&gt;DNA&lt;/em&gt; or &lt;em&gt;protein&lt;/em&gt; GLAM2 motifs&lt;span class=&quot;sample_output&quot;&gt; (&lt;a href=&quot;../doc/examples/glam2scan_example_output_files/glam2scan.html&quot;&gt;sample output&lt;/a&gt; for &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/At.glam2&#39;&gt;GLAM2 motif&lt;/a&gt; and &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/At.faa&#39;&gt;sequences&lt;/a&gt;)&lt;/span&gt;.&lt;span class=&#39;manual_link&#39;&gt; See this &lt;a href=&quot;../doc/glam2scan.html?man_type=web&quot;&gt;Manual&lt;/a&gt; for more information.&lt;/span&gt;  &lt;footer&gt;&lt;br&gt;&lt;br&gt;GLAM2Scan finds matches, in a set of sequences, to motifs discovered by GLAM2. Each match receives a score, indicating how well it fits the motif.  The GLAM2 motifs may be in either plain text (&lt;code&gt;glam2.txt&lt;/code&gt;) or HTML (&lt;code&gt;glam2.html&lt;/code&gt;) format.&lt;/footer&gt;">GLAM2Scan</a></div></div><div class="menu_topic" data-info="Compare query motif(s) to known motifs." data-id="4"><div class="menu_arrow closed">▶</div><div class="menu_arrow open">▼</div>Motif Comparison</div><div class="menu_subtopics" style="max-height: 0px;"><div><a href="https://meme-suite.org/meme/tools/tomtom" class="menu_subtopic" data-info="&lt;dfn&gt;Tomtom&lt;/dfn&gt; compares one or more motifs against a database of known motifs (e.g., JASPAR). Tomtom will rank the motifs in the database and produce an alignment for each significant match&lt;span class=&quot;sample_output&quot;&gt; (&lt;a href=&quot;../doc/examples/tomtom_example_output_files/tomtom.html&quot;&gt;sample output&lt;/a&gt; for &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/STRGGTCAN.meme&#39;&gt;motif&lt;/a&gt; and JASPAR CORE 2014 database)&lt;/span&gt;.&lt;span class=&#39;manual_link&#39;&gt; See this &lt;a href=&quot;../doc/tomtom.html?man_type=web&quot;&gt;Manual&lt;/a&gt; for more information.&lt;/span&gt;  &lt;footer&gt;&lt;br&gt;&lt;br&gt;Tomtom searches one or more query motifs against one or more databases of target motifs (and their reverse complements when applicable), and reports for each query a list of target motifs, ranked by &lt;i&gt;p&lt;/i&gt;-value.  The &lt;i&gt;E&lt;/i&gt;-value and the &lt;i&gt;q&lt;/i&gt;-value of each match is also reported.  The &lt;i&gt;q&lt;/i&gt;-value is the minimal false discovery rate at which the observed similarity would be deemed significant.  The output contains results for each query, in the order that the queries appear in the input file.  &lt;br&gt;&lt;br&gt;The web version of Tomtom provides a large number of target motif databases, or you can upload your own motifs in &lt;a href=meme-format.html&gt;MEME Motif Format&lt;/a&gt;, or enter them by typing in many additional formats.  &lt;br&gt;&lt;br&gt;For a given pair of motifs, the program considers all offsets between the motifs, while requiring a minimum number of overlapping positions. For a given offset, each overlapping position is scored using one of seven column similarity functions defined below. Columns in the query motif that don&#39;t overlap the target motif are assigned a score equal to the median score of the set of random matches to that column. &lt;br&gt;&lt;br&gt;In order to compute the scores, Tomtom needs to know the frequencies of the letters of the sequence alphabet in the database being searched (the &#39;background&#39; letter frequencies). By default, the background letter frequencies included in the query motif file are used. The scores of columns that overlap for a given offset are summed. This summed score is then converted to a &lt;i&gt;p&lt;/i&gt;-value. The reported &lt;i&gt;p&lt;/i&gt;-value is the minimal &lt;i&gt;p&lt;/i&gt;-value over all possible offsets.  To compensate for multiple testing, each reported &lt;i&gt;p&lt;/i&gt;-value is converted to an &lt;i&gt;E&lt;/i&gt;-value by multiplying it by twice the number of target motifs.  As a second type of multiple-testing correction, &lt;i&gt;q&lt;/i&gt;-values for each match are computed from the set of &lt;i&gt;p&lt;/i&gt;-values and reported.&lt;/footer&gt;">Tomtom</a></div></div><div class="menu_topic" data-info="Predict regulatory links." data-id="5"><div class="menu_arrow closed">▶</div><div class="menu_arrow open">▼</div>Gene Regulation</div><div class="menu_subtopics" style="max-height: 0px;"><div><a href="https://meme-suite.org/meme/tools/tgene" class="menu_subtopic" data-info="&lt;dfn&gt;T-Gene&lt;/dfn&gt; predicts &lt;em&gt;regulatory links&lt;/em&gt; between each of the &lt;em&gt;loci&lt;/em&gt; (genomic locations, typically TF ChIP-seq peak locations) that you provide and each of the organism&#39;s &lt;em&gt;genes&lt;/em&gt;&lt;span class=&quot;sample_output&quot;&gt; (&lt;a href=&quot;../doc/examples/tgene_example_output_files/tgene.html&quot;&gt;sample output&lt;/a&gt; for &lt;a href=&#39;https://meme-suite.org/meme/doc/examples/example-datasets/P300.chr21.bed&#39;&gt;these loci on Chr21&lt;/a&gt;)&lt;/span&gt;.&lt;span class=&quot;manual_link&quot;&gt; See this &lt;a href=&quot;../doc/tgene.html?man_type=web&quot;&gt;Manual&lt;/a&gt; or this &lt;a href=&quot;../doc/tgene-tutorial.html&quot;&gt;Tutorial&lt;/a&gt; for more information.&lt;/span&gt;  &lt;footer&gt;&lt;br&gt;&lt;br&gt;T-Gene predicts the regulatory targets of each of your loci based its distance from a potential target TSS, and, if a &quot;tissue panel&quot; is provided, based on the correlation across the tissues of the level of a histone modification at the locus with the expression of the transcript.  Both proximity to the gene and high correlation between a locus and a gene is evidence of that the locus may be involved in the regulation of the gene. The web version of T-Gene provides you with a number of genomes and tissue panels.&lt;/footer&gt;">T-Gene</a></div></div><div class="menu_topic" data-info="&lt;dfn&gt;BED2FASTA&lt;/dfn&gt; &lt;em&gt;extracts genomic regions&lt;/em&gt; specified in a BED file you provide from the associated genome&lt;span class=&quot;sample_output&quot;&gt; (&lt;a href=&quot;../doc/examples/bed2fasta_example_output_files/Rxra.fa&quot;&gt;sample output&lt;/a&gt; using this &lt;a href=&quot;../doc/examples/example-datasets/Rxra.bed&quot;&gt;BED file&lt;/a&gt; and the &lt;a href=&#39;https://meme-suite.org/meme/db/sequences&#39;&gt;UCSC hg19 human genome&lt;/a&gt;)&lt;/span&gt;." data-id="6"><div class="menu_arrow closed">▶</div><div class="menu_arrow open">▼</div>Utilities</div><div class="menu_subtopics" style="max-height: 0px;"><div><a href="https://meme-suite.org/meme/tools/bed2fasta" class="menu_subtopic" data-info="&lt;dfn&gt;BED2FASTA&lt;/dfn&gt; &lt;em&gt;extracts genomic regions&lt;/em&gt; specified in a BED file you provide from the associated genome&lt;span class=&quot;sample_output&quot;&gt; (&lt;a href=&quot;../doc/examples/bed2fasta_example_output_files/Rxra.fa&quot;&gt;sample output&lt;/a&gt; using this &lt;a href=&quot;../doc/examples/example-datasets/Rxra.bed&quot;&gt;BED file&lt;/a&gt; and the &lt;a href=&#39;https://meme-suite.org/meme/db/sequences&#39;&gt;UCSC hg19 human genome&lt;/a&gt;)&lt;/span&gt;.">BED2FASTA</a></div></div><div class="menu_topic" data-info="Read help pages for all the tools in the MEME Suite. This documentation can help you understand how the tools work and how to best use them.  Each help page describes the tool when run via the web, and also describes the inputs and outputs of the tool when you run it from the command line after installing MEME Suite on your computer." data-id="7"><div class="menu_arrow closed">▶</div><div class="menu_arrow open">▼</div>Manual</div><div class="menu_subtopics" style="max-height: 0px;"><div><a href="https://meme-suite.org/meme/doc/overview.html?man_type=web" class="menu_subtopic" data-info="See help pages for all the tools in the MEME Suite.">OVERVIEW</a><div class="menu_divider">Motif Discovery</div><a href="https://meme-suite.org/meme/doc/meme.html?man_type=web" class="menu_subtopic" data-info="Read documentation on using MEME.">MEME</a><a href="https://meme-suite.org/meme/doc/streme.html?man_type=web" class="menu_subtopic" data-info="Read documentation on using STREME.">STREME</a><a href="https://meme-suite.org/meme/doc/xstreme.html?man_type=web" class="menu_subtopic" data-info="Read documentation on using XSTREME.">XSTREME</a><a href="https://meme-suite.org/meme/doc/meme-chip.html?man_type=web" class="menu_subtopic" data-info="Read documentation on using MEME-ChIP.">MEME-ChIP</a><a href="https://meme-suite.org/meme/doc/glam2.html?man_type=web" class="menu_subtopic" data-info="Read documentation on using GLAM2.">GLAM2</a><a href="https://meme-suite.org/meme/doc/momo.html?man_type=web" class="menu_subtopic" data-info="Read documentation on using MoMo.">MoMo</a><a href="https://meme-suite.org/meme/doc/dreme.html?man_type=web" class="menu_subtopic" data-info="Read documentation on using DREME.">DREME (deprecated)</a><div class="menu_divider">Motif Enrichment</div><a href="https://meme-suite.org/meme/doc/sea.html?man_type=web" class="menu_subtopic" data-info="Read documentation on using SEA.">SEA</a><a href="https://meme-suite.org/meme/doc/centrimo.html?man_type=web" class="menu_subtopic" data-info="Read documentation on using CentriMo.">CentriMo</a><a href="https://meme-suite.org/meme/doc/ame.html?man_type=web" class="menu_subtopic" data-info="Read documentation on using AME.">AME</a><a href="https://meme-suite.org/meme/doc/spamo.html?man_type=web" class="menu_subtopic" data-info="Read documentation on using SpaMo.">SpaMo</a><a href="https://meme-suite.org/meme/doc/gomo.html?man_type=web" class="menu_subtopic" data-info="Read documentation on using GOMo.">GOMo</a><div class="menu_divider">Motif Scanning</div><a href="https://meme-suite.org/meme/doc/fimo.html?man_type=web" class="menu_subtopic" data-info="Read documentation on using FIMO.">FIMO</a><a href="https://meme-suite.org/meme/doc/mast.html?man_type=web" class="menu_subtopic" data-info="Read documentation on using MAST.">MAST</a><a href="https://meme-suite.org/meme/doc/mcast.html?man_type=web" class="menu_subtopic" data-info="Read documentation on using MCAST.">MCAST</a><a href="https://meme-suite.org/meme/doc/glam2scan.html?man_type=web" class="menu_subtopic" data-info="Read documentation on using GLAM2Scan.">GLAM2Scan</a><div class="menu_divider">Motif Comparison</div><a href="https://meme-suite.org/meme/doc/tomtom.html?man_type=web" class="menu_subtopic" data-info="Read documentation on using Tomtom.">Tomtom</a><div class="menu_divider">Gene Regulation</div><a href="https://meme-suite.org/meme/doc/tgene.html?man_type=web" class="menu_subtopic" data-info="Read documentation on using T-Gene.">T-Gene</a><div class="menu_divider">Utilities</div><a href="https://meme-suite.org/meme/doc/bed2fasta.html?man_type=web" class="menu_subtopic" data-info="Read documentation on using BED2FASTA.">BED2FASTA</a></div></div><div class="menu_topic" data-info="Find out more about how to use the MEME Suite." data-id="8"><div class="menu_arrow closed">▶</div><div class="menu_arrow open">▼</div>Guides &amp; Tutorials</div><div class="menu_subtopics" style="max-height: 0px;"><div><a href="https://meme-suite.org/meme/doc/ame-tutorial.html?man_type=web" class="menu_subtopic" data-info="Learn how to use AME to perform motif enrichment analysis on a set of sequences.">AME</a><a href="https://meme-suite.org/meme/doc/centrimo-tutorial.html?man_type=web" class="menu_subtopic" data-info="Learn how to use CentriMo to perform positional motif enrichment analysis on a set of sequences.">CentriMo</a><a href="https://meme-suite.org/meme/doc/dreme-tutorial.html?man_type=web" class="menu_subtopic" data-info="Learn how to use DREME to find short nucleotide motifs.">DREME (deprecated)</a><a href="https://meme-suite.org/meme/doc/fimo-tutorial.html?man_type=web" class="menu_subtopic" data-info="Learn how to use FIMO to find individual matches to each of your motifs in a set of sequences.">FIMO</a><a href="https://meme-suite.org/meme/doc/glam2_tut.html?man_type=web" class="menu_subtopic" data-info="Learn how to use GLAM2 to find GAPPED motifs.">GLAM2</a><a href="https://gt-scan.csiro.au/gt-scan/docs/manual" class="menu_subtopic" data-info="Learn how to use GT-Scan to identify good targets for CRISPR/Cas and other genome targeting technologies.">GT-Scan</a><a href="http://www.nature.com/nprot/journal/v9/n6/full/nprot.2014.083.html" class="menu_subtopic" data-info="Learn how to analyze large DNA or RNA datasets using MEME-ChIP. It can analyze peak regions identified by ChIP-seq, cross-linking sites identified by CLIP-seq and related assays, as well as sets of genomic regions selected using other criteria. MEME-ChIP performs de novo motif discovery, motif enrichment analysis, motif location analysis and motif clustering, providing a comprehensive picture of the DNA or RNA motifs that are enriched in the input sequences.">MEME-ChIP</a><a href="https://meme-suite.org/meme/doc/streme-tutorial.html?man_type=web" class="menu_subtopic" data-info="Learn how to use STREME to discover motifs that are enriched in the primary dataset that you provide. (If you have fewer than 50 sequences, MEME may work better than STREME.)  STREME accepts sets of sequences in the DNA, RNA, Protein alphabets, or in a custom alphabet designed by you. You can also use STREME to discover motifs that are enriched in the primary dataset relative to a second (control) dataset that you provide.">STREME</a><a href="https://meme-suite.org/meme/doc/tgene-tutorial.html?man_type=web" class="menu_subtopic" data-info="Learn how to use T-Gene to predict regulatory links between regulatory elements and genes.">T-Gene</a></div></div><div class="menu_topic" data-info="See samples of the output produced by MEME Suite programs." data-id="9"><div class="menu_arrow closed">▶</div><div class="menu_arrow open">▼</div>Sample Outputs</div><div class="menu_subtopics" style="max-height: 0px;"><div><div class="menu_divider">Motif Discovery</div><a href="https://meme-suite.org/meme/doc/examples/meme_example_output_files/meme.html?man_type=web" class="menu_subtopic" data-info="See sample output from MEME.">MEME Sample</a><a href="https://meme-suite.org/meme/doc/examples/streme_example_output_files/streme.html?man_type=web" class="menu_subtopic" data-info="See sample output from STREME.">STREME Sample</a><a href="https://meme-suite.org/meme/doc/examples/xstreme_example_output_files/xstreme.html?man_type=web" class="menu_subtopic" data-info="See sample output from XSTREME.">XSTREME Sample</a><a href="https://meme-suite.org/meme/doc/examples/memechip_example_output_files/meme-chip.html?man_type=web" class="menu_subtopic" data-info="See sample output from MEME-ChIP.">MEME-ChIP Sample</a><a href="https://meme-suite.org/meme/doc/examples/glam2_example_output_files/glam2.html?man_type=web" class="menu_subtopic" data-info="See sample output from GLAM2.">GLAM2 Sample</a><a href="https://meme-suite.org/meme/doc/examples/momo_example_output_files/momo.html?man_type=web" class="menu_subtopic" data-info="See sample output from MoMo.">MoMo Sample</a><a href="https://meme-suite.org/meme/doc/examples/dreme_example_output_files/dreme.html?man_type=web" class="menu_subtopic" data-info="See sample output from DREME.">DREME Sample (deprecated)</a><div class="menu_divider">Motif Enrichment</div><a href="https://meme-suite.org/meme/doc/examples/sea_example_output_files/sea.html?man_type=web" class="menu_subtopic" data-info="See sample output from SEA.">SEA Sample</a><a href="https://meme-suite.org/meme/doc/examples/centrimo_example_output_files/centrimo.html?man_type=web" class="menu_subtopic" data-info="See sample output from CentriMo.">CentriMo Sample</a><a href="https://meme-suite.org/meme/doc/examples/ame_example_output_files/ame.html?man_type=web" class="menu_subtopic" data-info="See sample output from AME.">AME Sample</a><a href="https://meme-suite.org/meme/doc/examples/spamo_example_output_files/spamo.html?man_type=web" class="menu_subtopic" data-info="See sample output from SpaMo">SpaMo Sample</a><a href="https://meme-suite.org/meme/doc/examples/gomo_example_output_files/gomo.html?man_type=web" class="menu_subtopic" data-info="See sample output from GOMo.">GOMo Sample</a><div class="menu_divider">Motif Scanning</div><a href="https://meme-suite.org/meme/doc/examples/fimo_example_output_files/fimo.html?man_type=web" class="menu_subtopic" data-info="See sample output from FIMO.">FIMO Sample</a><a href="https://meme-suite.org/meme/doc/examples/mast_example_output_files/mast.html?man_type=web" class="menu_subtopic" data-info="See sample output from MAST.">MAST Sample</a><a href="https://meme-suite.org/meme/doc/examples/mcast_example_output_files/mcast.html?man_type=web" class="menu_subtopic" data-info="See sample output files from MCAST.">MCAST Sample</a><a href="https://meme-suite.org/meme/doc/examples/glam2scan_example_output_files/glam2scan.html?man_type=web" class="menu_subtopic" data-info="See sample output files from GLAM2Scan.">GLAM2Scan Sample</a><div class="menu_divider">Motif Comparison</div><a href="https://meme-suite.org/meme/doc/examples/tomtom_example_output_files/tomtom.html?man_type=web" class="menu_subtopic" data-info="See sample output files from Tomtom.  ">Tomtom Sample</a><div class="menu_divider">Gene Regulation</div><a href="https://meme-suite.org/meme/doc/examples/tgene_example_output_files/tgene.html?man_type=web" class="menu_subtopic" data-info="See sample output files from T-Gene.  ">T-Gene Sample</a></div></div><div class="menu_topic" data-info="Read about the file formats that the MEME Suite uses." data-id="10"><div class="menu_arrow closed">▶</div><div class="menu_arrow open">▼</div>File Format Reference</div><div class="menu_subtopics" style="max-height: 0px;"><div><a href="https://meme-suite.org/meme/doc/alphabet-format.html?man_type=web" class="menu_subtopic" data-info="A custom alphabet can be specified to MEME Suite tools using this custom alphabet definition format.">*Alphabet Format</a><a href="https://meme-suite.org/meme/doc/bed-format.html?man_type=web" class="menu_subtopic" data-info="BED Genomic Loci format is a format used by some tools in the MEME Suite for identifying genomic regions.">*BED Loci Format</a><a href="https://meme-suite.org/meme/doc/fasta-format.html?man_type=web" class="menu_subtopic" data-info="FASTA sequence format is the main sequence format read by tools in the MEME Suite.">*FASTA Sequence</a><a href="https://meme-suite.org/meme/doc/meme-format.html?man_type=web" class="menu_subtopic" data-info="MEME motif format is the main motif format read by tools in the MEME Suite.">*MEME Motif format</a><a href="https://meme-suite.org/meme/doc/ame-output-format.html?man_type=web" class="menu_subtopic" data-info="AME produces an HTML results file, a TSV results file, and a TSV file containing the names of all sequences classified as &#39;positive&#39;.">AME output formats</a><a href="https://meme-suite.org/meme/doc/centrimo-output-format.html?man_type=web" class="menu_subtopic" data-info="CentriMo produces an HTML results file, a TSV results file, and a text file (suitable for plotting) of the number of sites in each window for each motif.">CentriMo output formats</a><a href="https://meme-suite.org/meme/doc/fimo-output-format.html?man_type=web" class="menu_subtopic" data-info="FIMO produces an HTML results file, a TSV results file and a GFF3 file containing all the significant matches for each motif.">FIMO output formats</a><a href="https://meme-suite.org/meme/doc/gomo-output-format.html?man_type=web" class="menu_subtopic" data-info="GOMo produces an HTML results file, a TSV results file and an XML file">GOMo output formats</a><a href="https://meme-suite.org/meme/doc/mcast-output-format.html?man_type=web" class="menu_subtopic" data-info="MCAST produces an HTML results file, a TSV results file and a GFF3 file containing all the significant matches to sets of motifs.">MCAST output formats</a><a href="https://meme-suite.org/meme/doc/meme-chip-output-format.html?man_type=web" class="menu_subtopic" data-info="MEME-ChIP produces an HTML results file, a TSV file containing a summary of its results, and a file containing all significant motifs in MEME motif format.">MEME-ChIP output formats</a><a href="https://meme-suite.org/meme/doc/momo-output-format.html?man_type=web" class="menu_subtopic" data-info="MoMo produces an HTML results file, a TSV results file, a file of motifs in MEME format, and PNG sequence logo files.">MoMo output formats</a><a href="https://meme-suite.org/meme/doc/sea-output-format.html?man_type=web" class="menu_subtopic" data-info="SEA produces an HTML results file, and a TSV results file.">SEA output formats</a><a href="https://meme-suite.org/meme/doc/spamo-output-format.html?man_type=web" class="menu_subtopic" data-info="SpaMo produces an HTML results file and a TSV results file.">SpaMo output formats</a><a href="https://meme-suite.org/meme/doc/streme-output-format.html?man_type=web" class="menu_subtopic" data-info="STREME produces an HTML results file, a text file containing the discovered motifs in MEME motif format, and a TSV containing the names of sequences containing each motif.">STREME output formats</a><a href="https://meme-suite.org/meme/doc/tgene-output-format.html?man_type=web" class="menu_subtopic" data-info="T-Gene produces an HTML results file, a Gene Targets TSV file, a Gene Elements TSV file, a TSS Targets TSV file, a TSS Elements TSV file and a Links TSV file.">T-Gene output formats</a><a href="https://meme-suite.org/meme/doc/tomtom-output-format.html?man_type=web" class="menu_subtopic" data-info="Tomtom produces an HTML results file, a TSV results file, and a XML file containing all the significant matches for each motif.">Tomtom output formats</a><a href="https://meme-suite.org/meme/doc/xstreme-output-format.html?man_type=web" class="menu_subtopic" data-info="XSTREME produces an HTML results file, a TSV file containing a summary of its results, and a file containing all non-redundant, &lt;i&gt;ab initio&lt;/i&gt; motifs discovered by XSTREME in MEME motif format.">XSTREME output formats</a><a href="https://meme-suite.org/meme/doc/dmix-format.html?man_type=web" class="menu_subtopic" data-info="Dirichlet mixture priors capture the tendency of certain letters (e.g., protein residues) to align with each other. These priors allow the user to bias the search for motifs by MEME and GLAM2.">Dirichlet Mixtures</a><a href="https://meme-suite.org/meme/doc/fasta-coordinates-format.html?man_type=web" class="menu_subtopic" data-info="FASTA sequence coordinates (e.g., genomic) in FASTA sequence headers.">FASTA Coordinates</a><a href="https://meme-suite.org/meme/doc/psm-format.html?man_type=web" class="menu_subtopic" data-info="Peptides identified from tandem mass spectra.">Peptide-Spectrum Match</a><a href="https://meme-suite.org/meme/doc/bfile-format.html?man_type=web" class="menu_subtopic" data-info="Background letter frequencies can be specified to MEME Suite tools using this Markov model based format.">Markov Background Model</a><a href="https://meme-suite.org/meme/doc/psp-format.html?man_type=web" class="menu_subtopic" data-info="Priors (weights) on each position in each input sequence can be specified to MEME by a file in PSP format. These priors allow the user to bias the search for motifs by MEME.">Position-specific prior (PSP)</a><a href="https://meme-suite.org/meme/doc/overview.html?man_type=web#formats" class="menu_subtopic" data-info="See the complete list of formats that are documented in the MEME Suite.">Other Supported Formats</a></div></div><div class="menu_topic" data-info="View information about the databases installed on this server." data-id="11"><div class="menu_arrow closed">▶</div><div class="menu_arrow open">▼</div>Databases</div><div class="menu_subtopics" style="max-height: 0px;"><div><a href="https://meme-suite.org/meme/db/motifs" class="menu_subtopic" data-info="View information on the motif databases installed on this server.">Motif Databases</a><a href="https://meme-suite.org/meme/db/sequences" class="menu_subtopic" data-info="View information on the sequence databases installed on this server.">Sequence Databases</a><a href="https://meme-suite.org/meme/db/gomo" class="menu_subtopic" data-info="View information on the GOMo sequence databases installed on this server.">GOMo Sequence Databases</a><a href="https://meme-suite.org/meme/db/tgene" class="menu_subtopic" data-info="View information on the T-Gene Genomes and Histone/Expression Tissue Panels installed on this server.">T-Gene Genomes and Tissue Panels</a></div></div><div class="menu_topic" data-info="Download a copy of the MEME Suite software and databases for running on your own computer." data-id="12"><div class="menu_arrow closed">▶</div><div class="menu_arrow open">▼</div>Download &amp; Install</div><div class="menu_subtopics" style="max-height: 0px;"><div><a href="https://meme-suite.org/doc/download.html" class="menu_subtopic" data-info="Download a copy of the MEME Suite for local installation, as well as motif databases and GOMo databases which can be used with it.">Download MEME Suite and Databases</a><a href="https://meme-suite.org/meme/doc/copyright.html?man_type=web" class="menu_subtopic" data-info="Read the copyright notice.">Copyright Notice</a><a href="https://meme-suite.org/meme/doc/install.html?man_type=web" class="menu_subtopic" data-info="Learn how to install the MEME Suite.">Installation Guide</a><a href="https://meme-suite.org/meme/doc/release-notes.html?man_type=web" class="menu_subtopic" data-info="Find out what was added to the MEME Suite in this and previous releases.">Release Notes</a><a href="http://techtransfer.universityofcalifornia.edu/NCD/20911.html" class="menu_subtopic" data-info="Acquire a license for commercial use.">Commercial Licensing</a></div></div><div class="menu_topic" data-info="Get help using the MEME Suite." data-id="13"><div class="menu_arrow closed">▶</div><div class="menu_arrow open">▼</div>Help</div><div class="menu_subtopics" style="max-height: 0px;"><div><a href="https://groups.google.com/forum/#!forum/meme-suite" class="menu_subtopic" data-info="Ask questions on the forums.">Q&amp;A Forum</a><a href="mailto:meme-suite@uw.edu" class="menu_subtopic" data-info="Send the webmaster an email. This link will open your email client">Email Webmaster</a><a href="mailto:meme-suite@uw.edu" class="menu_subtopic" data-info="Send the developers an email. This link will open your email client">Email Developers</a></div></div><div class="menu_topic" data-info="View the available MEME Suite servers." data-id="14"><div class="menu_arrow closed">▶</div><div class="menu_arrow open">▼</div>Alternate Servers</div><div class="menu_subtopics" style="max-height: 0px;"><div><a href="https://meme-suite.org/meme" class="menu_subtopic" data-info="Go to the main MEME Suite site.">Main Server</a></div></div><div class="menu_topic" data-info="Find out who contributed to the MEME Suite and how to cite the individual tools in publications." data-id="15"><div class="menu_arrow closed">▶</div><div class="menu_arrow open">▼</div>Authors &amp; Citing</div><div class="menu_subtopics" style="max-height: 0px;"><div><a href="https://meme-suite.org/meme/doc/authors.html?man_type=web" class="menu_subtopic" data-info="The authors of the MEME Suite.">Authors</a><a href="https://meme-suite.org/meme/doc/cite.html?man_type=web" class="menu_subtopic" data-info="How to cite the MEME Suite or individual tools from the MEME Suite.">Citing the MEME Suite</a><a href="https://meme-suite.org/meme-software/usage-reports" class="menu_subtopic" data-info="Data on past usage of the MEME Suite via the production server at https://meme-suite.org.">MEME Suite Usage Statistics</a></div></div><div class="menu_topic" data-info="The recent jobs you have run from this web browser." data-id="16"><div class="menu_arrow closed">▶</div><div class="menu_arrow open">▼</div>Recent Jobs</div><div class="menu_subtopics" style="max-height: 0px;"><div><div class="menu_clear_jobs">Clear All</div></div></div><a class="menu_previous" href="https://meme-suite.org/meme/meme_5.5.4">↪ Previous version 5.5.4</a></div>
    <!-- page begins -->
    <noscript>
      <div class="no_script">
        <div class="no_script_warning">
        <h1>Javascript is disabled! &#9785;</h1>
        <p>The MEME Suite web application requires the use of JavaScript but
        Javascript doesn't seem to be available on your browser.</p>
        <p>Please re-enable Javascript to use the MEME Suite.</p>
        </div>
      </div>
    </noscript>
    
    
    <div class="prog_logo">
      <img src="./Job Status - MEME Suite_files/memechip_icon.png" alt="MEME-ChIP Logo">
      <h1>MEME-ChIP</h1>
      <h2>Motif Analysis of Large Nucleotide Datasets</h2>
    </div>
    
    <p class="done" id="message">
    <span class="info expired">The job you requested could not be found. If you
    have followed a link in your verification email it is probable that the
    job has expired and the output has been deleted to clear up
    resources.</span>
    <span class="info pending">Please wait. Your MEME-ChIP job is now queued 
    awaiting available resources.  
    You may bookmark this page or use the <b>Recent Jobs</b> menu at the left
    to access your job's results.
    </span>
    <span class="info active">Please wait. Your MEME-ChIP job is now running.
    Further details may be available below.
    You may bookmark this page or use the <b>Recent Jobs</b> menu at the left
    to access your job's results.
    </span>
    <span class="info suspended">Please wait. Your MEME-ChIP job has been temporally
    suspended and is awaiting resources to continue.
    You may bookmark this page or use the <b>Recent Jobs</b> menu at the left
    to access your job's results.
    </span>
    <span class="info failed">An error occurred running your MEME-ChIP job. Further
    details may be available below.</span>
    <span class="info done">Your MEME-ChIP job is complete. The results should be
    displayed below.</span>
    <span class="info unknown">The status code received is unknown.</span>
    </p>
    
    <div class="" tabindex="0" id="details">
      <h3>Job Details</h3>
      <span class="collapsed"> ...</span>
      <span class="expanded"> ▼</span>
      <script>
      // create scope to avoid polluting global scope
      (function() {
        "use strict";
        var btn;
        btn = $("details");
        btn.addEventListener("click", function() {
          toggle_class(btn, 'expanded');
          resize_preview();
        });
        btn.addEventListener("keypress", function(evt) {
          if (evt.which == 32 || evt.keyCode == 32) {
            toggle_class(btn, 'expanded');
            resize_preview();
          }
        });
      })();
      </script>
    </div>
    <div id="info">
      <p>No job details available to display.</p>
    </div>
    <div id="preview_container">
    <iframe id="IF_1717648509839" src="./Job Status - MEME Suite_files/index.html" style="height: 479.636px;"></iframe>
    </div>
    <script>
    (function() {
      "use strict";
      var job, container;
      container = $("info");
      if ((job = get_job(id)) != null) {
        display_job_information(container, report["items"], job);
      }
    })();
    </script>
  

</body></html>